components/src/dynamo/vllm/tests/test_vllm_worker_handler.py

```
@@ -346,11 +346,61 @@ async def fake_generate(*args, **kwargs):
+    async def test_generate_tokens_routed_experts_start_echoes_prompt_start(self):
+        """routed_experts.start echoes SamplingParams.routed_experts_prompt_start
+        (the offset vLLM trimmed) so the RL consumer can align the completion."""
+        from vllm.sampling_params import SamplingParams
+
+        # routed_experts_prompt_start is an RL-patch field; set it post-construct
+        # so the test runs against stock vLLM too (the worker reads it via
+        # getattr, defaulting to 0 when absent).
+        sampling_params = SamplingParams(max_tokens=1)
+        try:
+            sampling_params.routed_experts_prompt_start = 5
+        except Exception:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)